controller/src/util/pod_watcher.rs

```diff
@@ -109,593 +104,571 @@ fn get_broker_pod_owner_kind(pod: &Pod) -> BrokerPodOwnerKind {
-                    &instance_name
-                );
-                return Ok(());
-            }
-        };
-        let instance_uid = instance
+    if let Some(configuration_service_spec) = &configuration.spec.configuration_service_spec {
+        let configuration_uid = configuration
             .metadata
             .uid
             .as_ref()
-            .ok_or_else(|| anyhow::anyhow!("UID not found for instance: {}", instance_name))?;
-        self.add_instance_and_configuration_services(
-            &instance_name,
-            instance_uid,
+            .context("UID not found for configuration")?;
+        let ownership = OwnershipInfo::new(
+            OwnershipType::Configuration,
+            configuration_name.to_string(),
+            configuration_uid.clone(),
+        );
+        let mut labels: BTreeMap<String, String> = BTreeMap::new();
+        labels.insert(
+            AKRI_CONFIGURATION_LABEL_NAME.to_string(),
+            configuration_name.to_string(),
+        );
+        let app_name = create_service_app_name(configuration_name);
+        let config_svc = create_new_service_from_spec(
+            &app_name,
             namespace,
-            &configuration_name,
-            &configuration,
-            kube_interface,
-        )
-        .await?;
-
-        Ok(())
+            ownership.clone(),
+            configuration_service_spec,
+            labels,
+        )?;
+        // TODO: handle already exists error
+        api.apply(config_svc, &ctx.identifier).await?;
     }
+    Ok(())
+}
 
-    /// This creates new service or updates existing service with ownership.
-    #[allow(clippy::too_many_arguments)]
-    async fn create_or_update_service(
-        &self,
-        instance_name: &str,
-        configuration_name: &str,
-        namespace: &str,
-        label_name: &str,
-        label_value: &str,
-        ownership: OwnershipInfo,
-        service_spec: &ServiceSpec,
-        is_instance_service: bool,
-        kube_interface: &impl KubeInterface,
-    ) -> anyhow::Result<()> {
-        trace!(
-            "create_or_update_service - instance={:?} with ownership:{:?}",
-            instance_name,
-            &ownership
-        );
+pub fn create_new_service_from_spec(
+    app_name: &str,
+    svc_namespace: &str,
+    ownership: OwnershipInfo,
+    svc_spec: &ServiceSpec,
+    mut labels: BTreeMap<String, String>,
+) -> anyhow::Result<Service> {
+    labels.insert(APP_LABEL_ID.to_string(), app_name.to_owned());
+    labels.insert(CONTROLLER_LABEL_ID.to_string(), API_NAMESPACE.to_string());
+    let owner_references: Vec<OwnerReference> = vec![OwnerReference {
+        api_version: ownership.get_api_version(),
+        kind: ownership.get_kind(),
+        controller: ownership.get_controller(),
+        block_owner_deletion: ownership.get_block_owner_deletion(),
+        name: ownership.get_name(),
+        uid: ownership.get_uid(),
+    }];
```

Comment:
It would be easier to implement `From<OwnershipInfo>` for `OwnerReference` than repeating this block every time